src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -1,9 +1,87 @@
+(* genesis_ledger_helper.ml *)
+
+(* for consensus nodes, read download ledger, proof, and constants from file, or
+     download from S3
+   for nonconsensus nodes, download genesis proof only; for native code, can
+     load from file or S3; for Javascript code, load only from S3
+*)
+
+[%%import
+"/src/config.mlh"]
+
+[%%ifdef
+consensus_mechanism]
+
 open Core
 open Async
-open Coda_base
+
+[%%else]
+
+[%%if
+ocaml_backend = "native"]
+
+open Core
+open Async
+
+[%%elif
+ocaml_backend = "js_of_ocaml"]
+
+open Core_kernel
+open Async_kernel
+
+[%%else]
+
+[%%error
+"Unsupported OCaml backend"]
+
+[%%endif]
+
+module Coda_base = Coda_base_nonconsensus
+module Cache_dir = Cache_dir_nonconsensus.Cache_dir
+
+[%%endif]
 
 type exn += Genesis_state_initialization_error
 
+let s3_root = "https://s3-us-west-2.amazonaws.com/snark-keys.o1test.net/"
+
+let proof_filename_root = "genesis_proof"
+
+[%%if
+ocaml_backend = "native"]
+
+let copy_file ~logger ~filename ~genesis_dir ~src_dir ~extract_target =
```

Comment:
The name `genesis_dir` here (and elsewhere) isn't really appropriate any more, makes this a little confusing. Can we rename it to `genesis_identifier` or similar?